docs/core/whats-new/dotnet-9/libraries.md

```diff
@@ -184,22 +271,318 @@ The following code snippet shows some example usage.
+
+```csharp
+string result = string.Join(", ", new string[3] { "a", "b", "c" });
+string result = string.Join(", ", "a", "b", "c");
+```
+
+Prior to .NET 9, when you pass the values individually, the C# compiler emits code identical to the first call by producing an implicit array around the three arguments.
+
+Starting in C# 13, you can use `params` with any argument that can be constructed via a collection expression, including spans (<xref:System.Span%601> and <xref:System.ReadOnlySpan%601>). That's beneficial for a variety of reasons, including performance. The C# compiler can store the arguments on the stack, wrap a span around them, and pass that off to the method, which avoids the implicit array allocation that would have otherwise resulted.
+
+.NET 9 now includes over 60 methods with a `params ReadOnlySpan<T>` parameter. Some are brand new overloads, and some are existing methods that already took a `ReadOnlySpan<T>` but now have that parameter marked with `params`. The net effect is if you upgrade to .NET 9 and recompile your code, you'll see performance improvements without making any code changes. That's because the compiler prefers to bind to span-based overloads than to the array-based overloads.
+
+For example, `String.Join` now includes the following overload, which implements the new pattern: <xref:System.String.Join(System.String,System.ReadOnlySpan{System.String})?displayProperty=nameWithType>
+
+Now, a call like `string.Join(", ", "a", "b", "c")` is made without allocating an array to pass in the `"a"`, `"b"`, and `"c"` arguments.
+
+## System.Numerics
+
+The following changes have been made in the <xref:System.Numerics> namespace:
+
+- [BigInteger upper limit](#biginteger-upper-limit)
+- [`BigMul` APIs](#bigmul-apis)
+- [Vector conversion APIs](#vector-conversion-apis)
+- [Vector create APIs](#vector-create-apis)
+- [Additional acceleration](#additional-acceleration)
+
+### BigInteger upper limit
+
+<xref:System.Numerics.BigInteger> supports representing integer values of essentially arbitrary length. However, in practice, the length is constrained by limits of the underlying computer, such as available memory or how long it would take to compute a given expression. Additionally, there exist some APIs that fail given inputs that result in a value that's too large. Because of these limits, .NET 9 enforces a maximum length of `BigInteger`, which is that it can contain no more than `(2^31) - 1` (approximately 2.14 billion) bits. Such a number represents an almost 256 MB allocation and contains approximately 646.5 million digits. This new limit ensures that all APIs exposed are well behaved and consistent while still allowing numbers that are far beyond most usage scenarios.
```

Comment:
Seems like an opportunity to link to a breaking change notice. We should ensure one gets created if there isn't one yet.

Reply:
https://github.com/dotnet/docs/issues/41752